src/slack-listener.ts

```diff
@@ -146,22 +151,77 @@ const simple = new SimpleReceiver((bolt) => {
             }
         })
     })
-    bolt.event('channel_left', async ({event, client, logger}) => {
+    bolt.event('channel_left', async ({event, client}) => {
         if (await channelMembershipEventAlreadyHandled(event)) {
             return
         }
         await withBotId(client, async (botId) => {
             await removeChannel(botId, event.channel)
         })
     })
-    bolt.event('group_left', async ({event, client, logger}) => {
+    bolt.event('group_left', async ({event, client}) => {
         if (await channelMembershipEventAlreadyHandled(event)) {
             return
         }
         await withBotId(client, async (botId) => {
             await removeChannel(botId, event.channel)
         })
     })
+    bolt.action('configure_emoji', async ({action, ack, respond}) => {
+        const actionResult = await handleConfigureEmojiAction(action)
+        let response: SlackBolt.RespondArguments = {
+            text: "There may have been an error, but I'm not sure. Try to confirm whether your action took effect.",
+            response_type: "ephemeral",
+        }
+        if (actionResult !== undefined) {
+            response = {
+                replace_original: true,
+                blocks: [
+                    {
+                        type: 'section',
+                        text: {
+                            type: 'mrkdwn',
+                            text: emojiConfigurePrompt(actionResult.image),
+                        }
+                    },
+                    {
+                        type: 'context',
+                        elements: [
+                            {
+                                type: 'mrkdwn',
+                                text: `Success: ${actionResult.image} ${emojiDescriptions[actionResult.emoji]}`,
+                            }
+                        ]
+                    },
+                ],
+            }
+        }
+        await respond(response)
+        await ack()
+    })
+    // bolt.command(...) handles slash commands. We'll set up just single command, with subcommands within that.
+    // The command name itself is configured within the Slack app settings (https://api.slack.com/apps/), and is out
+    // of our control in this code base.
+    //
+    // In other words, even though we accept any command, we actually expect that there'll just be a single command
+    // that always gets used for any given instance of this app. We just don't know what it is.
```

Comment:
Oh, I like that! I saw you approved, so I'll just upload that and then squash 'em all.

Just to  check, the regex should be `/github-notifier|.*/`, right? (`|` not space)

Reply:
the ` .*` was me guessing that you get the whole command string there so `/github-notifier emoji :rocket:`. Mostly in _not_ using `|` and requiring the first bit to be `/github-notifier` (opinionated). I'm fine either way or without, hence the approval.

Reply:
Oh, got it. So firstly, the regex is only for the actual command name, not the args. But I'd prefer to keep the command name unopinionated in the code, so that admins can change it just via an app configuration setting (on Slack's site), rather than in the code.

In particular, I actually have two instances of this bot running — the KlothoBot we all know, and then a KlothoBot-Yuval that I use for development (it listens to a couple stubbed repos I set up for the purpose, and it posts to a private channel with just me and it). I'd prefer to not need to modify this source to maintain that second bot.

Reply:
That reminds me — I need to update the manifest json for this change!